shopinvader_customer_multi_user/models/res_partner.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019 Camptocamp SA (http://www.camptocamp.com).
+# @author Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import random
+import string
+
+from odoo import api, fields, models
+
+
+def _generate_token(length=8):
+    return "".join(
+        random.choice(string.ascii_letters + string.digits)
+        for __ in range(length)
+    )
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    invader_user_token = fields.Char(readonly=True, index=True)
+    type = fields.Selection(
```

Comment:
@simahawk `type `is a reserved keyword....

Reply:
it's not mine... It's the odoo core's field :sob: